File: hw/astudillo/U3/HW16Maze/maze_python/src/maze/controller/door.py
from .room import Room


class Door:
    def __init__(self, first_room: Room, second_room: Room):
        self._first_room = first_room
        self._second_room = second_room

    @property
    def first_room(self) -> Room:
        return self._first_room

    @property
    def second_room(self) -> Room:
        return self._second_room

    def __str__(self) -> str:
        return f"Door connecting {self._first_room.coordinate} and {self._second_room.coordinate}"

File: hw/astudillo/U3/HW16Maze/maze/src/main/java/org/example/edu/ec/espe/maze/controller/Door.java
package org.example.edu.ec.espe.maze.controller;

public class Door {
    private final Room firstRoom;
    private final Room secondRoom;

    public Door(Room firstRoom, Room secondRoom) {
        this.firstRoom = firstRoom;
        this.secondRoom = secondRoom;
    }

    public Room getFirstRoom() {
        return firstRoom;
    }

    public Room getSecondRoom() {
        return secondRoom;
    }
}
